src/main/scala/is/hail/expr/ir/Interpret.scala

```diff
@@ -610,21 +610,114 @@ object Interpret {
             initOps()(region, rvAggs, globals, false)
           }
 
-          value.rvd.aggregate[Array[RegionValueAggregator]](rvAggs)({ case (rvaggs, rv) =>
+          val zval = rvAggs
+          val combOp = { (rvAggs1: Array[RegionValueAggregator], rvAggs2: Array[RegionValueAggregator]) =>
+            rvAggs1.zip(rvAggs2).foreach { case (rvAgg1, rvAgg2) => rvAgg1.combOp(rvAgg2) }
+            rvAggs1
+          }
+          value.rvd.crdd.cmapPartitions { (ctx, it) =>
```

Comment:
this breaks the RVD abstraction boundary, which I guess I'm OK with (we seem to not have not settled on a good abstraction given that the boundary is often broken). Did you consider / see an easy way to expose an aggregation where instead of a seqop we have a partition op?

Reply:
That does sound better

Reply:
Maybe rvd.aggregateWithPartitionOp?

Reply:
And the seq op is `(PC, U, T) => U`?

Reply:
And the seq op is `(PC, U, T) => U`?

Reply:
And the seq op is `(PC, U, T) => U`?

Reply:
Yeah that actually seems a closer fit and similar to things we've done before. I had imagined the following, but the thing you've proposed seems better because the user doesn't need to write `it.foreach { rv => ... }`.

```scala
// in RVD
def aggregatePartitions[U](
  zeroValue: U,
  partitionOp: (RVDContext, U, Iterator[RegionValue]) => U,
  combOp: (U, U) => U
): U

// in ContextRDD
def aggregatePartitions[U, T](
  zeroValue: U,
  partitionOp: (C, U, Iterator[T]) => U,
  combOp: (U, U) => U
): U
```